src/store/users-list/actions.tsx

```diff
@@ -1,22 +1,53 @@
-import {actionTypes, IUser} from './types';
-import {doFetch} from '../../utils';
+import { actionTypes, IUser } from './types';
+import { IState } from '../';
+import { doFetch, payload} from '../../utils';
 
 interface IAction {
   type: actionTypes,
-  payload?: IUser[] | IUser,
+  payload?: {
+    users: IUser[] | IUser | [],
+    isUsersLoading: boolean,
+    isUsersFail: boolean,
+  } | IUser
 };
 
 type IDispatch = (arg: IAction) => (IAction);
 
-const SET_USERS = (users: IUser[]) => ({type: actionTypes.SET_USERS, payload: users});
+const SET_USERS_LOADING = (users: IUser[]) => ({
+  type: actionTypes.SET_USERS_LOADING, 
+  payload: {
+    users,
+    isUsersLoading: true,
+    isUsersFail: false,
+}});
+
+const SET_USERS_SUCCESS = (users: IUser[]) => ({
+  type: actionTypes.SET_USERS_SUCCESS, 
+  payload: {
+    users,
+    isUsersLoading: false,
+    isUsersFail: false,
+}});
+
+const SET_USERS_FAIL = (users: IUser[]) => ({
+  type: actionTypes.SET_USERS_FAIL, 
+  payload: {
+    users,
+    isUsersLoading: false,
+    isUsersFail: true,
+}});
+
 const SET_SELECTED_USER = (user: IUser) => ({type: actionTypes.SET_SELECTED_USER, payload: user});
 
-export const getUsers = (payload: Object) => async (dispatch: IDispatch) => {
+export const getUsers = () => async (dispatch: IDispatch, getState: () => IState) => {
   try{
+    dispatch(SET_USERS_LOADING([]));
```

Comment:
Could we create some generic function, that will receive the action name that should be triggered as a parameter

Reply:
`const generateAsyncActions = (type) => ({
  REQUEST: `${type}_REQUEST`,
  SUCCESS: `${type}_SUCCESS`,
  FAIL: `${type}_FAIL`,
});
const {REQUEST, SUCCESS, FAIL} = generateAsyncActions(actionType);
dispatch(someGenericFunction(REQUEST, PAYLOAD))